plugins/telemetry/stats_poller.go

```diff
@@ -0,0 +1,118 @@
+package telemetry
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"git.fd.io/govpp.git/api"
+	"github.com/gogo/status"
+	"github.com/ligato/cn-infra/logging"
+	"google.golang.org/grpc/codes"
+
+	"github.com/ligato/vpp-agent/api/configurator"
+	"github.com/ligato/vpp-agent/api/models/vpp"
+	vpp_interfaces "github.com/ligato/vpp-agent/api/models/vpp/interfaces"
+	"github.com/ligato/vpp-agent/plugins/telemetry/vppcalls"
+)
+
+type statsPollerServer struct {
+	handler vppcalls.TelemetryVppAPI
+
+	log logging.Logger
+}
+
+func (s *statsPollerServer) PollStats(req *configurator.PollStatsRequest, svr configurator.StatsPoller_PollStatsServer) error {
+	var pollSeq uint32
+
+	if req.PeriodSec == 0 {
+		return status.Error(codes.InvalidArgument, "PeriodSec must be greater than 0")
+	}
+	period := time.Duration(req.PeriodSec) * time.Second
+
+	tick := time.NewTicker(period)
+	defer tick.Stop()
+
+	s.log.Debugf("starting to poll stats every %v", period)
+	for {
+		pollSeq++
+		s.log.WithField("seq", pollSeq).Debugf("polling stats..")
+
+		vppStatsCh := make(chan vpp.Stats)
+		var vppStatsErr error
+		go func() {
+			vppStatsErr = s.streamVppStats(vppStatsCh)
+		}()
+		for vppStats := range vppStatsCh {
```

Comment:
looks like streamVppStats method can exit before closing channel. Isn't this a problem?

Reply:
did I found a DEADLOCK :scream: